### ISSUE TITLE
ABCDeGraphGenerator assertion error

### ISSUE THREAD

**Issue by bartoszpankratz** [bkamins/ABCDGraphGenerator.jl#8]
For small _n_  (e.g. 1e5) parallelized code is often throwing an assertion error: 
`(AssertionError("2 * length(global_edges) == length(stubs)")`
it is especially visible for ξs smaller than 0.7:
![plot](https://user-images.githubusercontent.com/32309411/125059143-4c5b1880-e0ab-11eb-94d1-2744aebd8d15.png)
In attachement I am sending all the logs:
[logz.zip](https://github.com/bkamins/ABCDGraphGenerator.jl/files/6790420/logz.zip)
and example configuration:
```julia
seed = "33"                   # RNG seed, use "" for no seeding
n = "100000"                   # number of vertices in graph
t1 = "2.1"                      # power-law exponent for degree distribution
d_min = "1"                   # minimum degree
d_max = "316"                  # maximum degree sqrt(n)
d_max_iter = "10000"           # maximum number of iterations for sampling degrees
t2 = "1.1"                      # power-law exponent for cluster size distribution
c_min = "500"                  # minimum cluster size Int(0.005*n)               
c_max = "20000"                # maximum cluster size Int(0.2*n)
c_max_iter = "10000"           # maximum number of iterations for sampling cluster sizes
# Exactly one of xi and mu must be passed as Float64. Also if xi is provided islocal must be set to false or omitted.
xi = "0.55"                    # fraction of edges to fall in background graph
#mu = "0.2"                   # mixing parameter
islocal = "false"             # if "true" mixing parameter is restricted to local cluster, otherwise it is global
isCL = "false"                # if "false" use configuration model, if "true" use Chung-Lu
degreefile = "deg.dat"        # name of file do generate that contains vertex degrees
communitysizesfile = "cs.dat" # name of file do generate that contains community sizes
communityfile = "com.dat"     # name of file do generate that contains assignments of vertices to communities
networkfile = "edge.dat"      # name of file do generate that contains edges of the generated graph
```

**Comment by bkamins** [comment]
@bartoszpankratz - we wanted to keep track of the problems in @tolcz repository (I am not sure he is watching this repo, so I am pinging him).

**Comment by bartoszpankratz** [comment]
@bkamins  Yeah, I know, but I still don't get a link to new repo and on @tolcz fork issues are not enabled.

**Comment by tlcz** [comment]
Issues are now enabled in https://github.com/tolcz/ABCDeGraphGenerator.jl

**Comment by tlcz** [comment]
Hi @bartoszpankratz
Thank you for your time and effort testing package [ABCDe](https://github.com/tolcz/ABCDeGraphGenerator.jl).
I tried to reproduce the assertion error with the latest version of the package available from from https://github.com/tolcz/ABCDeGraphGenerator.jl (master) using the provided configuration and any number of threads from 2 to 12, but I could not. 
Could you please make sure you are running this version of the package and verify if the exception reappears?
If so, could you please call `ABCDeGraphGenerator.gen_graph()` directly, in a similar way it is done in the script `utils/abcd_sampler.jl`? This will produce a complete stack trace of an exception which in turn will allow me to determine the root of the problem. Once it is done and exception confirmed, could you please open a new issue at https://github.com/tolcz/ABCDeGraphGenerator.jl/issues where it belongs.

**Comment by tlcz** [comment]
@bartoszpankratz I don't know what data you depicted on the chart, nevertheless for your convenience I added [two assertions](https://github.com/tolcz/ABCDeGraphGenerator.jl/commit/41f6aa252910ff0ef31dd56fc9a1d3504a119d78) into `utils/abcd_sampler.jl`to validate if configuration model output follows its input configuration.
